### PR TITLE
BENCH scaling of a GridSearch with n_jobs

### PR DESCRIPTION
**Pull request by tomMoral**
Add a benchmark to compare `joblib` scaling for a `GridSearch` on synthetic data.
This benchmark allows to compare 2 versions of `joblib` by running it multiple times.

image

### CODE REVIEW

**Review by GaelVaroquaux — APPROVED** [review]
Very nice!

The bump between 6 and 8 jobs is quite curious.

File: benchmarks/bench_grid_search_scaling.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    )
+    parser.add_argument(
+        "--name", "-n", type=str, default="",
+        help="Name to save the results with. This can be used to compare "
+        "different branches with '-c'."
+    )
+    parser.add_argument(
+        "--compare", "-c", action="append",
+        help="Compare the results with existing results saved with name equal "
+        "to the argument value."
```
Comment (tomMoral) [inline benchmarks/bench_grid_search_scaling.py:42]:
```suggestion
        help="Loads the results from a benchmark saved previously with a name "
        "given as the present argument value. This allows comparing the results "
        "across different runs of the benchmark and version of joblib."
```

**Comment by ogrisel** [comment]
Thanks, here is what I get on my laptop with 4 performance cores and 4 efficiency cores.

image

I cannot reproduce the plateauing effect in thomas' plot.

**Comment by ogrisel** [comment]
Here is the same result with a `ylim` to start at 0.

image

I find it less misleading.

**Comment by ogrisel** [comment]
Hum I forced pushed over my last commit to use a simple `plt.ylim(0, None)` instead of `plt.ylim(0, max_time)` with a manually computed `max_time` but github does not display the correct diff (yet).

**Comment by ogrisel** [comment]
I messed up with git, but I finally pushed a simpler `ylim`.

**Review by ogrisel — APPROVED** [review]

**Comment by ogrisel** [comment]
I think this kind of plots would be more informative as speed-ups factor vs a chosen baseline (e.g. the slowest or the single thread result from the current branch) but this is already useful as it is so I merged.

**Pull request merged**